doc/self-hosted/build.md

```diff
@@ -0,0 +1,60 @@
+# Build instructions for code-server 🏗️
+
+code-server has several yarn tasks for building the entire binary or specific portions of the app. This guide provides steps to do the aforementioned tasks. Coder recommends having Node 10.15.1+ installed.
+
+**NOTE:** Windows compatibility is in the works 
+> Yarn installation: https://yarnpkg.com/lang/en/docs/install
+
+### Build the binary
+1. Clone the [repository](https://github.com/codercom/code-server.git)
+2. `cd` into `code-server`
+3. Run `yarn`
+4. Run `yarn task build:server:binary`
+   > Binary will be placed in `packages/server` named after the platform and architecture (`cli-darwin-x64`)
+
+### Run without building
+1. Clone the [repository](https://github.com/codercom/code-server.git)
+2. `cd` into `code-server`
+3. Run `yarn`
+4. Run `yarn start`
+    > *If you run into issues with websocket disconnects, run with `yarn start --no-auth`*
+
+### Build Tasks
```

Comment:
I wonder if we shouldn't expose some of these commands. For example I'm not sure it's useful for anyone to run `server:binary:package:copy`. Or as another example, we have `vscode:install` and `vscode:patch` but building the bootstrap fork for example does both of those anyway (and `vscode:patch` also runs `vscode:install`), so running them separately might not be useful.